### ISSUE TITLE
Crash in Charts v2.2.5-Swift3.0

### ISSUE THREAD

**Issue by madhikarma** [danielgindi/Charts#1570]
Hi there 

I found a crash at runtime using the Chart2.2.5-Swift3.0 branch

In certain circumstances the CGPoint created in the `lowestVisibleXIndex` method returns NaN (see screenshot below)

![screen shot 2016-09-27 at 16 00 54](https://cloud.githubusercontent.com/assets/1456685/18879327/beaf4078-84cb-11e6-8cf9-7dd9d2d207a1.png)

![screen shot 2016-09-27 at 16 00 52](https://cloud.githubusercontent.com/assets/1456685/18879335/c6a75216-84cb-11e6-9ee8-2b61d8937ee6.png)

We were hoping to use with this version of the Charts API whilst we wait for the Charts 3 / Swift 3.0 release (and with that the new APIs for the library )

Can you advise about fixing the problem or whether its safe to use the branch or if we need to Charts 3?

Thank you for your time and for this library

**Comment by madhikarma** [comment]
...using Xcode 8.0 GM / iOS 10 SDK and running on the iPhone 5s Simulator

**Comment by liuxuan30** [comment]
giving the code you post, `pt` is just the left bottom corner, which should be a valid value. I suspect something wrong with the transfomer or your setup.

Could you debug more? e.g. print out `viewPortHandler.contentRect`, print out `pt` value before transfrom, print out the transformer `pixelToValueMatrix`, print out `pt` after transform.

if `pt` is really the valid left bottom corner point coordinates, and after the transform, it is NaN?

Also, it's better it can be produced by ChartsDemo so we could debug. But for now, you have to debug on your side.

Never seen this

**Comment by petester42** [comment]
since we have no plans to officially support swift 3.0 with chart 2.x we should close all issues related to it and delete the branch to avoid confusion.

**Comment by madhikarma** [comment]
Makes sense, I thought it was worth trying the branch to keep the same Charts API but gain Swift 3.0 support.

I was able to patch the crash above with a NaN check inside the method but several more crashes occurred in other methods with the data set we were using so we stopped. 

The same data set with the v2.3.0 Swift 2.3 branch https://github.com/danielgindi/Charts/tree/v2.3.0 did not crash so we migrated to Swift 2.3 instead until Charts 3.0 Swift 3.0 is finalised.
